ngx-nuts-and-bolts-docs/docs/testing-utils/mock-storage.md

```diff
@@ -0,0 +1,25 @@
+---
+id: mock-storage
+title: Mock storage
+sidebar_label: Mock storage
+---
+
+When you need to interact with storage but still you are testing only a unit of the code, you can use `MockStorage` class for this purpose. Mocked version doesn't take into account stuff like size limits and other things taken care by `window.localstorage API` and just focuses on core functionality.
+
+## 1. Usage
+
+```ts
+let storage: Storage;
+
+beforeEach(async () => {
+	await TestBed.configureTestingModule({
+		providers: [
+			{
+				provide: Storage,
```

Comment:
I am not sure it's a good idea to use Storage as DI token. I expect most apps will create tokens like LOCAL_STORAGE and SESSION_STORAGE. In unit tests for a component that uses both local and session storage, testbed setup would probably look something like this:

```ts
{
  provide: LOCAL_STORAGE,
  useClass: MockStorage
},
{
  provide: SESSION_STORAGE,
  useClass: MockStorage
}
```